tests/testthat/test-add-variable.R

```diff
@@ -46,7 +46,17 @@ test_that("toVariable handles duplicate factor levels", {
             ))),
         "Duplicate factor levels given: disambiguating them in translation to Categorical type")
 })
-
+test_that("categoriesFromLevels parses levels correctly", {
+  expect_identical(
+    categoriesFromLevels(levels(iris$Species)),
+    list(structure(list(id = 1L, name = "setosa", numeric_value = 1L, missing = FALSE),     
+                   .Names = c("id", "name", "numeric_value", "missing")), 
+         structure(list(id = 2L, name = "versicolor", numeric_value = 2L, missing = FALSE), 
+                   .Names = c("id", "name", "numeric_value", "missing")), 
+         structure(list(id = 3L, name = "virginica", numeric_value = 3L, missing = FALSE),  
+                   .Names = c("id", "name", "numeric_value", "missing")))
+  )
```

Comment:
Tabs are 4 spaces, and could you simplify the test expectation by removing the `structure`s? named lists work fine and are more readable. 

For future reference, you can print out simplified list objects like this with `control=NULL`; see `?.deparseOpts`:

```
> cats <- crunch:::categoriesFromLevels(c("a", "b"))
> dput(cats)
list(structure(list(id = 1L, name = "a", numeric_value = 1L, 
    missing = FALSE), .Names = c("id", "name", "numeric_value", 
"missing")), structure(list(id = 2L, name = "b", numeric_value = 2L, 
    missing = FALSE), .Names = c("id", "name", "numeric_value", 
"missing")))
> dput(cats, control=NULL)
list(list(id = 1, name = "a", numeric_value = 1, missing = FALSE), 
    list(id = 2, name = "b", numeric_value = 2, missing = FALSE))
```

and control "keepInteger" may be all you need to get lists that compare identical:

```
> dput(cats, control="keepInteger")
list(list(id = 1L, name = "a", numeric_value = 1L, missing = FALSE), 
    list(id = 2L, name = "b", numeric_value = 2L, missing = FALSE))
```